### ISSUE TITLE
Demo parser is broken

### ISSUE THREAD

**Issue by briantbutton** [WebAssembly/wabt#2010]
Parser driving demo page is broken.  (https://webassembly.github.io/wabt/demo/wat2wasm/)  It doesn't parse many standard WABT examples.

~ It fails to parse several of its own sample code, including, for example 'sign-extension'.
~ It fails to parse its own sample code from about a year ago.  See below.
~ If fails to parse sample code given here:
    https://developer.mozilla.org/en-US/docs/WebAssembly/Reference/Control_flow/br

My sense is that commands like `get_local`, `set_local`, `get_global` and `set_global` have been dropped, killing backward compatibility.  

```(module
  (func $subtract (param i32 i32) (result i32)
    get_local 0       ;; 2
    get_local 1       ;; 5
    i32.sub)
)
```

**Comment by sbc100** [comment]
Yes, some of those instruction such as `get_local` were renamed to `local.get`.    I'm not sure its worth it for wabt to try to support the text names for those instruction. 

For the sign-extension stuff that could just be that the demo needs updating. sign-extension is enabled by default in newer versions of wabt but may not have been enabled by default when the demo was built.   We should probably re-built it (and also embed a version number on the page so we can see how old it is).

**Comment by sbc100** [comment]
It looks like the renaming was done back in https://github.com/WebAssembly/wabt/issues/933.  Support for the old names was kept around for a few years until #1792

**Comment by keithw** [comment]
> ~ It fails to parse several of its own sample code, including, for example 'sign-extension'.

Can you be more specific? I just tried the "sign extension" example and it seems to parse fine (after clicking the checkbox to enable the "sign extension" feature).

 ~ It fails to parse its own sample code from about a year ago. See below.

 ~ If fails to parse sample code given here: https://developer.mozilla.org/en-US/docs/WebAssembly/Reference/Control_flow/br
> 
> My sense is that commands like `get_local`, `set_local`, `get_global` and `set_global` have been dropped, killing backward compatibility.

As @sbc100 says, these commands were never part of the WebAssembly specification -- they are vestigial names that were renamed in 2018 before version 1.0 (the "MVP"). (This does seem to be a common issue -- see https://github.com/WebAssembly/wabt/issues/1802#issuecomment-1236657208, https://github.com/WebAssembly/wabt/issues/1883#issuecomment-1086705883, https://github.com/WebAssembly/wabt/issues/1820#issuecomment-1029690688.)

Perhaps you (or we) should write to MDN and ask them to update their code samples?

**Comment by briantbutton** [comment]
Thank you for your prompt and informed comments.  I updated by making changes to the syntax as suggested and it compiled.  I will close this issue on Monday.  I will contact Mozilla.

Compliments on your addition of the Base64 button.  It would have saved me fair time in years past.

I understand your spec-based defense of removing support for those instructions.  Let me offer my own POV on this.

Your demo parser page is my sole access to WASM.  I built many utilities using it, including this one.  (https://www.npmjs.com/package/md5-wasm)  The instructions shown there were the final authority on WASM.  I, like most programmers, don't go read the spec -- presumably that is your intent.

So, when your examples include things like get_local and get_global, and your parser parses them, that **_is_** WASM.  Breaking compatibility is a confidence killer.  And, spec notwithstanding, you are breaking backward compatibility with yourself.

I am all fixed up, thanks to you.  But there may be others like me.
